src/Core/File/LightData.hpp

```diff
@@ -43,54 +56,121 @@ class RA_CORE_API LightData : public AssetData {
+ * @{
+ */
+    /**
+     * Construct a directional light.
+     * A directional light is only defined by its color and its lighting direction.
+     * No attenuation on directional.
+     */
+    inline void setLight(const Core::Color &color, const Core::Vector3 &direction);
+
+    /**
+    * Construct a point light.
+    * A point light is defined by its color, its position and its attenuation.
     */
-#if 0
-    /// Acces to the data. The returned component must be attached to an entity after that.
-    // this will add a dependence on the core to the engine. Not a good idea ...
-    Ra::Engine::Light* getLight() const;
-#endif
+    inline void setLight(const Core::Color &color, const Core::Vector3 &position, LightAttenuation attenuation);
+
+    /**
+    * Construct a spot light.
+    * A spot light is defined by its color, its position, the cone axis and light distribution
+    * (constant inside inangle, quadratically deacreasing toward 0 fron inAngle to ouAngle) and its attenuation.
+    */
+    inline void setLight(const Core::Color &color, const Core::Vector3 &position, const Core::Vector3 &direction,
+                         Scalar inAngle, Scalar outAngle, LightAttenuation attenuation);
 
-    /// construct a directional light
-    inline void setLight( Core::Color color, Core::Vector3 direction );
-    /// construct a point light
-    inline void setLight( Core::Color color, Core::Vector3 position, LightAttenuation attenuation );
-    /// construct a spot light
-    inline void setLight( Core::Color color, Core::Vector3 position, Core::Vector3 direction,
-                          Scalar inangle, Scalar outAngle, LightAttenuation attenuation );
     /// construct an area light
-    inline void setLight( Core::Color color, LightAttenuation attenuation );
+    /**
+     * Construct a area light.
+     * An area light, (isotropic with constant emision defined by its color) is approximated by its center and
+     * the covariance matrices modelling the spatial extent as well as the elliptical distribution of normals.
+     */
+    inline void setLight( const Core::Color &color, const Core::Vector3 &cog, const Core::Matrix3 &spatialCov,
+                          const Core::Matrix3 &normalCov, LightAttenuation attenuation );
+/**@}*/
+
+/**
+ * \defgroup TypeOperators Helper functions to manage the type of the light.
+ * @{
+ */
+    /**
+    * Access to the type of the object
+    */
+    inline LightType getType() const;
 
-    /// QUERY
+    /**
+    * Modify the type of a light asset after its creation.
+    * \note Beware of the risk of data inconsistancy when using the operator. As a LightData object is a union like
+    * object, caller of this method must explicitely ensure a consistancy invariant over its object as the LightData
+    * class could not ensure such an invariant when this method is called.
+    * @param type the type to set.
+    */
+    inline void setType( const LightType& type );
```

Comment:
I understand this method can cause inconsistency, as it changes the accessing type of the union storing the light parameters.
So my question is: do we need it ? If there is no express need for this method I would prefer we remove it.

Reply:
We don't need these function, we can mark them as deprecated and removed progressively their uses from Asset importers and Light managers. For now, in the Radium distribution, they are used by ``AssimpLoader`` ecosystem and ``LightManager``. 
We can either uses the following (i.e ``isTypeLight``) but this will be less convenient for future extension. 
The better (that will require much more refactoring and code rewriting) is to rely on polymorphic factory (the same way the materials are managed through converter functions that ease the extension of supported material type) that will allow to extend the set of supported light sources in the engine from an external plugin, extension that is not feasible for now.

Reply:
Please mark `setType` with the `deprecated` attribute, and will properly fix this in v2.